### ISSUE TITLE
How to know, in PowerShell, whether an OnlineSipDomain value can be associated to an OnlineVoiceRoute 

### ISSUE THREAD

**Issue by vlethiec** [MicrosoftDocs/office-docs-powershell#9338]
Hi, 

The documentation regarding the "OnlinePstnGatewayList" attribute should contain more information regarding the accepted values. 
Two types of values are accepted: OnlinePSTNGateway and OnlineSipDomain. 

All the "OnlinePSTNGateway" values are authorized, but it is not the case for "OnlineSipDomain" values. 
Can you add some information to help us to identify accepted "OnlineSipDomain" values. 

![image](https://user-images.githubusercontent.com/52075388/165064229-579518f6-67ca-4f30-9bc5-717f98a17d00.png)

Thanks, 

Vincent
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 588f3a35-1526-dfaa-c94c-00f81935c70b
* Version Independent ID: 8061dab3-4913-f920-524b-892ac16e1f64
* Content: [New-CsOnlineVoiceRoute (SkypeForBusiness)](https://docs.microsoft.com/en-us/powershell/module/skype/new-csonlinevoiceroute?view=skype-ps)
* Content Source: [skype/skype-ps/skype/New-CsOnlineVoiceRoute.md](https://github.com/MicrosoftDocs/office-docs-powershell/blob/main/skype/skype-ps/skype/New-CsOnlineVoiceRoute.md)
* Service: **skypeforbusiness-powershell**
* GitHub Login: @tomkau
* Microsoft Alias: **tomkau**

**Comment by get-itips** [comment]
Hello @vlethiec Vincent, how are you?
Can you help me understand how did you conclude that OnlinePstnGatewayList accepts OnlineSipDomain?
Thank you!

**Comment by vlethiec** [comment]
Hi @get-itips Andres, 

The voice route can be associated to two different types of SBC. Classic hardware SBCs, which are defined as OnlinePSTNGateway in Teams PowerShell, and hosted SBCs, which are SBC hosted by techno vendors (or carrier), like Ribbon, Oracle or Audiocodes. These hosted SBCs are not defined as OnlinePSTNGateway in Teams PowerShell, but as OnlineSipDomain. 
You can find some explanation from the Oracle documentation below. 
https://www.oracle.com/a/otn/docs/oraclesbcwithteamshostingmodel.pdf

They explain several contraints : 

- "You cannot use the *.onmicrosoft.com tenant for the domain name."
- Use a subdomain
- Add a license to the user

My point is all the information come from the techno vendors and not from Microsoft. But according to my test, only SIP domains covering these criteria are accepted in the PSTN Gateway list. It means this behavior is hard coded in Teams and should be documented.  

I know it is a tricky point, but I would like to define my development on Microsoft documentation and not on SBC documentation. 

I would be happy to suggest some improvement on the documentation, but if I do so, for sure it must be double checked by the Teams PowerShell development team. 

To sum up, tell me Andres, if I should suggest some modification in the doc. 

Thanks in advance. 

Vincent

**Comment by get-itips** [comment]
Hello @vlethiec Vincent, thanks, I understand better now that you are talking about SBCs for multi tenant scenario and you are sitting from the customer side.

I'll check if an update is required to the article, thanks

**Comment by yogkumgit** [comment]
@get-itips

**Comment by yogkumgit** [comment]
@FilippSe Gentle reminder. Email sent.

**Comment by firatakandere** [comment]
Hi,
Only OnlinePSTNGateway values are accepted, OnlineSipDomain isn't checked during validation.

For multi-tenant setup;
One level higher fqdn will be checked if it exists as OnlinePSTNGateway.

For example;
Carrier SBC: `customers.adatum.biz`
Customer SBC: `sbc1.customers.adatum.biz`

Only Carrier SBC `customers.adatum.biz` should be created as OnlinePSTNGateway with New-CsOnlinePSTNGateway and
Customer SBC `sbc1.customers.adatum.biz` should be passed for `OnlinePstnGatewayList` in New-CsOnlineVoiceRoute.

The validation logic will look up for `customers.adatum.biz` in OnlinePstnGateway as well.

BR,
Firat.

**Comment by get-itips** [comment]
Thanks @firatakandere for the clarification

**Comment by yogkumgit** [comment]
@vlethiec Hope this comment is helpful for you. If you see a documentation update is required, please feel free to open an issue for the same. We proceed here to close it. Thanks for taking out some time to open the issue. Appreciate and encourage you to do the same in future also.
